doc/decisions/frontend_end_to_end_testing.md

```diff
@@ -0,0 +1,175 @@
+        self.search_term_input = page.locator('[aria-label="Enter your search term"]')
+
+    def navigate(self):
+        self.page.goto("https://bing.com")
+
+    def search(self, text):
+        self.search_term_input.fill(text)
+        self.search_term_input.press("Enter")
+```
+
+### Fast and easy prototyping of the tests
+
+Frontend tests can cost a lot of time and effort to develop.
+Python is a good fit to keep them simple and not overengineered, whilest creating them fast with every iteration.
+Given BDD and POM, it is allowing us to not only be simple but also readable, maintanable, explicit, etc.. ().
+Python fits perfectly with the above mentioned design decision and contributes to their philosphy (As long as we don't have performance issues or constraints).
+
+### Considered Alternatives
+
+#### Capture/Replay
+
+Because they are easily broken by changing elements on a webpage.
+
+#### Assert page by screenshots
+
+Too much effort and could be flaky. Also bad for testing user interaction.
+
+## Decision Language
+
+[Python](https://www.python.org) as programming language.
+
+### About the language:
+
+One would typically chose Java with Selenium for this testing approach but we went with python.
+[Why?](#fast-and-easy-prototyping-of-the-tests)
+It is also very newcomer friendly, easy and has a big and still growing community.
+
+### Considered Alternatives
+
+#### [Java](https://www.java.com/de/)
+
+Because you can do the same in python with less lines of code.
+
+#### [Javascript](https://www.javascript.com)
+
+Same as java.
+
+## Decision libraries
+
+- [Playwright for Python](https://playwright.dev/python/docs/intro)
+- [Pytest](https://docs.pytest.org/en/7.4.x/)
+- [Pytest-playwright](https://pypi.org/project/pytest-playwright/)
+- [Pytest-bdd](https://pypi.org/project/pytest-bdd/)
+- [mypy](https://mypy-lang.org)
+
+### About the libraries
+
+Most libraries offer more than we need.
+Some offer a bit less but it's mostly about:
+
+- Cross-browser
+- Cross-platform
+- Cross-language
+- Auto-wait
+- Tracing
+- Screenshots
+- Diagrams
+- Codegen/Recording
+- etc...
+
+What we need:
+
+- BDD support
+- Python binding
+- Headless browser
+
+What is nice to have:
+
+- Auto-wait
+- Screenshots
+- Diagrams
+
+Most of them would fit our constraints, so at the end it was between Selenium and Playwright, which had the biggest communities and overall great features.
+We chose Playwright because it allows us to get elements also by text and not only css selectors.
+Playwright has a very smooth installation and a superior documentation in comparison to selenium.
+We need headless browser support, so we can run the tests in our pipeline.
+
+### Considered Alternatives
+
+#### [Selenium](https://github.com/SeleniumHQ/selenium)
+
+Selenium is the older framework with the bigger community, but we decided against it, because we are already in the ecosystem of playwright, with our performance tests, and playwright is trending.
```

Comment:
These are very poor reasons. Please also add technical comparisons. Probably better if you use itemize.